src/test/java/com/google/sps/servlets/SearchServletTest.java

```diff
@@ -0,0 +1,279 @@
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+/** */
+public final class SearchServletTest {
+
+  private static final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  private SearchServlet searchObject;
+
+  @BeforeEach
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    searchObject = new SearchServlet();
+  }
+
+  @AfterEach
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Mock HttpServletRequest request;
+
+  @Test
+  /* Ensure the correct filters are set when all search parameters are specified. */
+  public void makeFiltersAllParams() {
+
+    request = Mockito.mock(HttpServletRequest.class);
+
+    when(request.getParameter("courseName")).thenReturn("CS 105");
+    when(request.getParameter("profName")).thenReturn("Smith");
+    when(request.getParameter("term")).thenReturn("Spring 2020");
+    when(request.getParameter("units")).thenReturn("1,2,3");
+
+    List<Filter> filters = searchObject.getFilters(request);
+
+    List<Filter> expectedFilters = new ArrayList<>();
+    List<Integer> units = new ArrayList<>();
+    units.add(1);
+    units.add(2);
+    units.add(3);
+    expectedFilters.add(new FilterPredicate("name", FilterOperator.EQUAL, "CS 105"));
+    expectedFilters.add(new FilterPredicate("professor", FilterOperator.EQUAL, "Smith"));
+    expectedFilters.add(new FilterPredicate("term", FilterOperator.EQUAL, "Spring 2020"));
+    expectedFilters.add(new FilterPredicate("units", FilterOperator.IN, units));
+
+    assertEquals(filters, expectedFilters);
+  }
+
+  @Test
+  /* Ensure no filters are made given no search parameters are specified. */
+  public void makeFiltersNoParams() {
+
+    request = Mockito.mock(HttpServletRequest.class);
+
+    // TODO: Find a better way to replicate null parameters -
+    // getting NullPointerException when trying other methods.
+    when(request.getParameter("courseName")).thenReturn("");
```

Comment:
make a function to make the request. I made one in my test file. Pretty sure it should for for you so feel free to copy it